File: machine/cpp/machine/memory/specs.cpp
#include "machine/memory/specs.h"

#include "machine/memory/map.h"
#include "machine/model_registry.h"

#include <iomanip>
#include <iostream>

namespace bmsx {

void configureRuntimeMemoryMap() {
	const uint32_t ramBytes = static_cast<uint32_t>(PSX_MACHINE_SPEC.ramBytes);
	const double ramMiB = static_cast<double>(ramBytes) / (1024.0 * 1024.0);
	const uint32_t dynamicRamBytes = ramBytes - MIN_RAM_SIZE;
	std::cerr
		<< "[RuntimeMemorySpecs] memory footprint: ram=" << ramBytes << " bytes ("
		<< std::fixed << std::setprecision(2) << ramMiB << " MiB) "
		<< "(io=" << IO_REGION_SIZE
		<< ", base_ram_used=" << BASE_RAM_USED_SIZE
		<< ", dynamic_ram=" << dynamicRamBytes
		<< ", geo_scratch=" << DEFAULT_GEO_SCRATCH_SIZE
		<< ")." << std::endl;
	configureMemoryMap(ramBytes);
}

} // namespace bmsx

File: machine/ts/machine/memory/specs.ts
import { PSX_MACHINE_SPEC } from '../model_registry';
import {
	DEFAULT_GEO_SCRATCH_SIZE,
	BASE_RAM_USED_SIZE,
	IO_REGION_SIZE,
	MIN_RAM_SIZE,
	configureMemoryMap,
} from './map';

export function configureRuntimeMemoryMap(): void {
	const ramBytes = PSX_MACHINE_SPEC.ramBytes;
	const footprintMiB = (ramBytes / (1024 * 1024)).toFixed(2);
	const dynamicRamBytes = ramBytes - MIN_RAM_SIZE;
	console.info(
		`memory footprint: ram=${ramBytes} bytes (${footprintMiB} MiB) `
		+ `(io=${IO_REGION_SIZE}, base_ram_used=${BASE_RAM_USED_SIZE}, dynamic_ram=${dynamicRamBytes}, `
		+ `geo_scratch=${DEFAULT_GEO_SCRATCH_SIZE}).`,
	);
	configureMemoryMap(ramBytes);
}
